File: src/sdk/php/3/core-classes/search-result/fetch-next/snippets/fetch-next-2.php

<?php

use \Kuzzle\Kuzzle;
use \Kuzzle\Document;
use \Kuzzle\Util\SearchResult;

function processDocument ($doc) {
  // do something with the document
}

$kuzzle = new Kuzzle('localhost');
$collection = $kuzzle->collection('collection', 'index');

try {
  $searchResult = $collection->search([
    'from' => 0,
    'size' => 1000,
    'scroll' => '30s',
    'filter' => []
  ]);
  
  while ($searchResult !== null) {
    foreach ($searchResult->getDocuments() as $doc) {
      procesDocument($doc);
    }
    
    $searchResult = $searchResult->fetchNext();
  }
}
catch (ErrorException $e) {
  // Handle errors here
}

File: src/sdk/android/3/core-classes/search-result/fetch-next/snippets/fetch-next-2.java

import io.kuzzle.sdk.core.Kuzzle;
import io.kuzzle.sdk.core.Options;

Kuzzle kuzzle = new Kuzzle("localhost");

JSONObject filter = new JSONObject();

Options options = new Options();
options.setFrom((long) 0);
options.setSize((long) 1000);
options.setScroll("30s");

ResponseListener<SearchResult> listener = new ResponseListener<SearchResult>() {
  @Override
  public void onSuccess(SearchResult searchResult) {
    if (searchResult == null) {
      return;
    }

    for (Document doc : searchResult.getDocuments()) {
      // do something with the document
      // this.processDocument(doc);
    }

    searchResult.fetchNext(this);
  }

  @Override
  public void onError(JSONObject error) {
    // handle errors here
  }
};

kuzzle
  .collection("collection", "index")
  .search(filter, options, listener);
